### DISCUSSION TITLE
How to create Artifactory instance using only access token?

### DISCUSSION THREAD

**Discussion by dr4kk4r** [anancarv/python-artifactory#133] [Q&A] [answered]
From what I can tell currently Artifactory class only allows for authorization via user name and password/api key.

How can I use access token instead?

**Comment by mohan-barathi** [comment]
Hi @dr4kk4r , 
did you manage to find an answer ? 
I have the same question now. 🙄

**Reply by anancarv** [reply]
Hello, currently we cannot use `access_token` for authorization. I'll create a new PR to allow this possibility.

**Comment by nymous** [comment, marked as answer]
As @anancarv said, PyArtifactory only supports the basic auth scheme, which doesn't work with an access token (I guess you mean the "Identity tokens" as JFrog calls them, or "Scoped tokens" in another screen).
However, if you generate a "Reference token" (starting with `cm` if I trust my testing), which isn't an Identity token (a JWT) but an opaque reference to an Identity token, it will work everywhere: in basic auth, in the `X-JFrog-Art-Api` header, or in the `Authorization: Bearer <token>` header.

Generating a Reference token can be done with the API by passing the `include_reference_token` parameter, in the Administration > Access tokens screen by checking the "Create Reference Token" checkbox, or when creating a token from your user profile (it always generates a Reference token there).
Source: https://jfrog.com/help/r/platform-api-key-deprecation-and-the-new-reference-tokens/what-is-a-reference-token

**Reply by mohan-barathi** [reply]
The Reference token is working as expected, just like user_name and password. 
Thank you @nymous 👍

**Comment by Ashwinmnr** [comment]
@dr4kk4r  This feature to use access token is now enabled by @anancarv on the latest version.
